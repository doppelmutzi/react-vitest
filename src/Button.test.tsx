import { render, fireEvent } from '@testing-library/react';
import { vi, describe, test, expect } from "vitest";
import Button from './Button';

describe('Button', () => {
  test('renders with correct text and handles click', () => {
    const handleClick = vi.fn();
    const { getByText } = render(<Button onClick={handleClick}>Click Me</Button>);
    
    const button = getByText('Click Me');
    fireEvent.click(button);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('matches snapshot', () => {
    const handleClick = vi.fn();
    const { container } = render(<Button onClick={handleClick}>Click Me</Button>);
    expect(container).toMatchSnapshot();
  });
});