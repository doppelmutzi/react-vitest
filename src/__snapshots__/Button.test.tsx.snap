// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > matches snapshot 1`] = `
<div>
  <button>
    Click Me
  </button>
</div>
`;
